### PR TITLE
chore: mark state and store packages as deprecated

### PR DESCRIPTION
**Pull request by melekes**
Please note that I didn't use the `Deprecated` keyword because doing this would result in many warnings for us and our users.

Follow-up to https://github.com/cometbft/cometbft/pull/2709

---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Review by adizere — APPROVED** [review]
Looks good, guess it makes sense to avoid using the magic key word.

File: state/doc.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// XXX: This package may be internalized (made private) in the future
```
Comment (cason) [inline state/doc.go:1]:
```suggestion
// XXX: This package may be internalized (made private) in future
```

Comment (cason) [reply state/doc.go:1]: The same in the next file. We can also make a single-line comment.

**Review by cason — APPROVED** [review]
Thank you for this. 

I don't understand the usage of `// XXX:` in terms of comment, but if you find it more appropriate, good.

**Pull request merged**